### ISSUE TITLE
jasmine.js:2219 Uncaught TypeError: this.expectationResultFactory is not a function

### ISSUE THREAD

**Issue by yrtimiD** [jasmine/jasmine#1147]
Version 2.4.1 fails on one specific test I have.
The error is:

```
jasmine.js:2219 Uncaught TypeError: this.expectationResultFactory is not a function
Suite.onException @ jasmine.js:2219
onException @ jasmine.js:2330
onException @ jasmine.js:1924
handleException @ jasmine.js:1928
attemptAsync @ jasmine.js:1918
QueueRunner.run @ jasmine.js:1871
(anonymous function) @ jasmine.js:1898
(anonymous function) @ jasmine.js:1842
onComplete @ jasmine.js:2467

```

After upgrading jasmine to the latest version (commit f6da084  on Mar 27), original error gone, but now I get a new one:

```
jasmine.js:1921 Uncaught TypeError: Cannot read property 'length' of undefined
QueueRunner.run @   jasmine.js:1921
QueueRunner.execute @   jasmine.js:1910
queueRunnerFactory  @   jasmine.js:706
Spec.execute    @   jasmine.js:362
fn  @   jasmine.js:2558
attemptAsync    @   jasmine.js:1967
QueueRunner.run @   jasmine.js:1922
(anonymous function)    @   jasmine.js:1949
...
```

The main problem I see is with error handling by jasmine - when this error happens jasmine doesn't reports failure to Karma but stops running other tests, making karma and CI think that all tests are passed:
`
Chrome 51.0.2704 (Windows 10 0.0.0): Executed 49 of 190 SUCCESS (0.149 secs / 0.074 secs)
`

**Comment by yrtimiD** [comment]
The error was caused by having beforeEach(..) within it(..) function.
I know it's stupid, but can happen with newbie devs and it's extremely hard to discover when you have a lot of tests.

**Comment by Gerg** [comment]
Hey @yrtimiD,

Glad to see you figured it out. Was there anything Jasmine could have done to make things easier to debug? What type of error message would you expect to see?

**Comment by seanhelvey** [comment]
I am receiving this error but I do not have a beforeEach within an it block. I created a new issue here https://github.com/jasmine/jasmine/issues/1156.

**Comment by derickbailey** [comment]
i'm also getting this error, and i do not have anything like a beforeEach in an it function. this is on a test suite that has worked fine for months, but i recently deleted the node_modules folder and re-installed everything. now i get this stack trace:

```
Fatal error: TypeError: this.expectationResultFactory is not a function
  at Suite.onException (/Users/derickbailey/dev/projects/rabbus/node_modules/grunt-jasmine-nodejs/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:2219:48)
  at QueueRunner.queueRunnerFactory.onException (/Users/derickbailey/dev/projects/rabbus/node_modules/grunt-jasmine-nodejs/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:2330:28)
  at onException (/Users/derickbailey/dev/projects/rabbus/node_modules/grunt-jasmine-nodejs/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1924:12)
  at [object Object]._onTimeout (/Users/derickbailey/dev/projects/rabbus/node_modules/grunt-jasmine-nodejs/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1910:11)
  at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by derickbailey** [comment]
i've been digging into this, and in my case the specific cause is having an `afterAll` with a `done` parameter that is never called.

created any test suite using node and jasmine-core (or whatever library on top of jasmine-core), and add this to your suite:

```js
afterAll(function(done){});
```

this will cause the above `Fatal error: TypeError: this.expectationResultFactory is not a function` error, instead of giving us a proper "timeout" error like older versions of jasmine used to do.

**Comment by slackersoft** [comment]
This sounds like it is probably a duplicate of #811 which should be fixed in the 2.5.x releases. Can you take a look at the new version and see if it fixes your issue?

Thanks for using Jasmine!
